pkg/internalapis/cnsvolumeoperationrequest/v1alpha1/cnsvolumeoperationrequest_types.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CnsVolumeOperationRequestSpec defines the desired state of CnsVolumeOperationRequest
+type CnsVolumeOperationRequestSpec struct {
+	// Name represents the name of the instance.
+	// There is no strict naming convention for instances; it is dependent on the caller.
+	Name string `json:"name,omitempty"`
+}
+
+// CnsVolumeOperationRequestStatus defines the observed state of CnsVolumeOperationRequest
+type CnsVolumeOperationRequestStatus struct {
+	// ID represents the VolumeID or SnapshotID of the backend volume or snapshot.
+	// Populated during successful CreateVolume or CreateSnapshot calls.
+	ID string `json:"id,omitempty"`
+	// Populated with the latest capacity on every successful ExtendVolume call for a volume.
+	Capacity int64 `json:"capacity,omitempty"`
+	// FirstOperationDetails stores the details of the first operation performed on the volume.
+	// For debugging purposes, clients should ensure that this information is never overwritten.
+	// More recent operation details should be stored in the LatestOperationDetails field.
+	FirstOperationDetails OperationDetails `json:"firstOperationDetails,omitempty"`
```

Comment:
I don't think we need FirstOperationDetails, we can easily debug based on the detail we have in LatestOperationDetails. I would suggest rename LatestOperationDetails to OperationDetails

Reply:
This came up in the design review. Its easier to debug if we clearly call out the first operation and then separately list newer operations.

Reply:
for debugging we use both CSI traces and VC logs, using available OpIDs and tasks we can easily find from Center logs which one was triggered first.

Reply:
I was thinking more on this. Can we use LatestOperationDetails to store FirstOperationDetails, and while cleaning up LatestOperationDetails to create space for more Ops, we can keep entry at index 0 where we have saved our FirstOperationDetails

Reply:
Yes that was the other proposal - keep the first entry static and then roll over the other 9 entries according to which is the latest one. The team thought that that approach is a little confusing to the user since there's no indication of which was first etc so we changed it to use separate fields.